cpp/src/parquet/statistics.cc

```diff
@@ -472,34 +474,40 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
     comparator_ = std::static_pointer_cast<TypedComparator<DType>>(comp);
     TypedStatisticsImpl::Reset();
     has_null_count_ = true;
-    has_distinct_count_ = true;
+    has_distinct_count_ = false;
```

Comment:
nit: has_distinct_count_ is set to false in `TypedStatisticsImpl::Reset();`

Reply:
Emmm currently writer **never** set `has_distinct_count_`, so I change it as disabled. In my opinion, user **should** explicit set `has_distinct_count_` when building, like:

```c++
FinishColumnChunk() {
  if(!fallback_) {
     set_distinct_count()
  }
}
```

The current `has_distinct_count_` is absolutely confusing...

Reply:
@pitrou As Gang mentioned here

1. In theory, `has_null_count_` is not added able if another statistics has `has_null_count_ = true`
2. However, in our implementions, builder will always set `has_null_count_ == true`

Maybe I can use the same way to treat them?

Reply:
Hmm... sorry, but I don't understand what you mean with "not added able".

Reply:
CMIW, `null_count_` can be safely accumulated and merged, so `has_null_count_` is set to true and `null_count_` is set to 0 at its initial stage. During the building process, `has_null_count_` should always be true unless something wrong happens.

On the contrary, `distinct_count_` cannot be merged and actually it is not set in the building process. It makes sense to set it to false at its initial stage.

I think @mapleFU means `distinct_count_` cannot be merged.

Reply:
Ok, fair enough.